src/pages/homeCollection/printLabels.js

```diff
@@ -77,26 +95,26 @@ export const getTotalAddressesToPrint = async () => {
 const generateParticipantCsvGetter = (name) => {
   const generateCsvButton = document.getElementById("generateCsv");
   if (generateCsvButton) {
-    generateCsvButton.addEventListener("click", () => {
+    generateCsvButton.addEventListener("click", async () => {
+      const totalAddressesLength = appState.getState().totalAddressesLength;
         const numberToPrint = document.getElementById("numberToPrint").value;
-          if (numberToPrint) {
-            const arrayLengthToProcess = appState.getState().totalAddressesLength
-            if (arrayLengthToProcess >= numberToPrint) {
-              const arrayToProcess = appState.getState().totalAddresses.filter(obj => obj.length !== 0)
-              const remainingArrayToProcess = arrayToProcess.slice(numberToPrint, appState.getState().totalAddressesLength)
-              appState.setState({totalAddresses: remainingArrayToProcess })
-              appState.setState({'totalAddressesLength': remainingArrayToProcess.length })
-              generateParticipantCsv(arrayToProcess.slice(0, numberToPrint));
+        if(!numberToPrint || !totalAddressesLength) {
+          triggerErrorModal(`No labels to print`);
+        } else if (numberToPrint > totalAddressesLength) {
+          triggerErrorModal(`Max labels to print: ${arrayLengthToProcess}`);
+        } else {
+          const totalAddressesRes = await getTotalAddressesToPrint(numberToPrint);
+          if (totalAddressesRes.code === 200) {
+              const arrayToProcess = totalAddressesRes.data;
+              appState.setState({'totalAddressesLength': totalAddressesLength - numberToPrint }); // No need for another API call
+              generateParticipantCsv(arrayToProcess);
               printLabelsTemplate(name);
               triggerSuccessModal('Success!');                 // Display success message
-            }
-            else if (appState.getState().totalAddressesLength === 0) {
-              triggerErrorModal(`No labels to print`)
-            }
-            else {
-              triggerErrorModal(`Max labels to print: ${arrayLengthToProcess}`);
-            } 
+          } else {
+            console.error('response', totalAddressesRes);
+            triggerErrorModal(`Error getting records; please review console.`);
```

Comment:
Just a nit here (this is very nice in-app communication):

Since this app isn't participant-facing, you can show a more detailed message in the error modal for the user to report or explain what the console is since many non-developers won't know what the 'console' is.